### PR TITLE
[Schema] Add title field to Resource and ResourceTemplate

### LINKED ISSUE #296: Resource and ResourceTemplate are missing the `title` field
## Summary

The MCP specification (2025-11-25) defines an optional `title` field for `Resource`
and `ResourceTemplate`, but the PHP SDK does not implement it.

## Spec definition

The [Resources specification](https://modelcontextprotocol.io/specification/2025-11-25/server/resources)
lists the following fields for `Resource`:

> * `uri`: Unique identifier for the resource
> * `name`: The name of the resource.
> * `title`: Optional human-readable name of the resource for display purposes.

Example from the spec:

```json
{
  "uri": "file:///project/src/main.rs",
  "name": "main.rs",
  "title": "Rust Software Application Main File",
  "description": "Primary application entry point",
  "mimeType": "text/x-rust"
}
```

`ResourceTemplate` follows the same pattern:

```json
{
  "uriTemplate": "file:///{path}",
  "name": "Project Files",
  "title": "📁 Project Files",
  "description": "Access files in the project directory"
}
```

All four server capability types in the spec define `title` as the optional
human-readable display label:

| Type               | Machine identifier | `title`              |
|--------------------|--------------------|----------------------|
| `Tool`             | `name`             | optional, implemented |
| `Prompt`           | `name`             | optional, implemented |
| `Resource`         | `uri`              | optional, **missing** |
| `ResourceTemplate` | `uriTemplate`      | optional, **missing** |

## Current SDK behaviour

`Mcp\Schema\Resource` and `Mcp\Schema\ResourceTemplate` have no `title` constructor
parameter, no `title` property, and do not include `title` in `jsonSerialize()`.
Servers that want to expose a human-readable label for a resource currently have no
way to do so.

Additionally, the docblock for `Resource::$name` currently reads:

> A human-readable name for this resource. This can be used by clients to populate
> UI elements.

This description conflicts with the spec, where `name` is a short identifier (e.g.
`"main.rs"`) and `title` is the human-readable display label.

## Proposed fix

Add `public readonly ?string $title = null` to both `Resource` and `ResourceTemplate`,
include it in `fromArray()` and `jsonSerialize()`, and update the `$name` docblock to
match the spec.

### PR DESCRIPTION
**Pull request by blackwell-systems**
## Summary

- Add the optional `title` field to `Resource` and `ResourceTemplate`, matching
  the MCP specification (2025-11-25) and the existing `Tool`/`Prompt` implementations
- Add `title` to `McpResource` and `McpResourceTemplate` attributes and wire it
  through `Discoverer`
- Add tests for deserialization, serialization, and null omission

## Context

The spec defines `title` as an optional human-readable display label for all four
server capability types. `Tool` and `Prompt` already have it; `Resource` and
`ResourceTemplate` were the only two missing.

| Type               | Machine identifier | `title`              |
|--------------------|--------------------|----------------------|
| `Tool`             | `name`             | implemented          |
| `Prompt`           | `name`             | implemented          |
| `Resource`         | `uri`              | **added in this PR** |
| `ResourceTemplate` | `uriTemplate`      | **added in this PR** |

The `$name` docblocks are updated to say "short identifier" instead of
"human-readable name", matching the spec's distinction between `name` and `title`.

Backward compatible: `$title` defaults to `null`, serialization omits it when null,
and all callers use named arguments.

Fixes #296

## Test plan

- [x] `ResourceTest::testTitleFromArray` verifies deserialization
- [x] `ResourceTest::testTitleSerialization` verifies output
- [x] `ResourceTest::testTitleOmittedWhenNull` verifies null omission
- [x] Same three tests for `ResourceTemplateTest`
- [x] Full suite: 744 tests, 2422 assertions, 0 failures

### CODE REVIEW

File: src/Capability/Attribute/McpResource.php [hunk 1]
```diff
@@ -24,18 +24,20 @@
...
+     * @param ?string               $title       optional human-readable title for display in UI
+     * @param ?string               $description optional description; defaults to class DocBlock summary
      * @param ?string               $mimeType    the MIME type, if known and constant for this resource
      * @param ?int                  $size        the size in bytes, if known and constant
-     * @param Annotations|null      $annotations optional annotations describing the resource
-     * @param ?Icon[]               $icons       Optional list of icon URLs representing the resource
-     * @param ?array<string, mixed> $meta        Optional metadata
+     * @param ?Annotations          $annotations optional annotations describing the resource
+     * @param ?Icon[]               $icons       optional icons representing the resource
+     * @param ?array<string, mixed> $meta        optional metadata
```
Comment (soyuka) [inline src/Capability/Attribute/McpResource.php:35]: not sure why all the changes, can't you just add the title?

**Review by soyuka — APPROVED** [review]
Except my small nitpicking comment lgtm

File: src/Capability/Attribute/McpResource.php (hunk 1 above)
Comment (blackwell-systems) [reply src/Capability/Attribute/McpResource.php:35]: The docblock changes align `$name` with the spec distinction between `name` (machine identifier) and `title` (display label). The old docblock said "human-readable name" which is now what `$title` is for. The other adjustments are just consistency: lowercase param descriptions and `Annotations|null` -> `?Annotations` to match the type declaration. Happy to revert the style changes to a separate commit if you prefer a minimal diff.

**Review by chr-hertel — APPROVED** [review]
